### PR TITLE
GOA-2251_Add_replacements_to_ontology

### PR DESCRIPTION
**Pull request by rantunesboreas**

### CODE REVIEW

File: indexing/src/main/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverter.java [hunk 1]
```diff
@@ -87,10 +86,13 @@
         return null;
     }
 
-    protected List<String> extractConsidersAsList(GenericTerm term) {
-        if (!isEmpty(term.consider())) {
-            return term.consider().stream()
-                    .map(GenericTerm::getId)
+    protected List<String> extractRelations(Collection<TermRelation> replaceList) {
```
Comment (eddturner) [inline indexing/src/main/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverter.java]: should this be called `extractReplacementTypes` or something?

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverterTest.java [hunk 2]
```diff
@@ -35,25 +34,91 @@ public void setup() {
 
     // considers
     @Test
-    public void extractsNoConsideredWhenNull() {
-        when(term.consider()).thenReturn(null);
+    public void extractsNoRelationsWhenNull() {
```
Comment (eddturner) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverterTest.java]: If method is renamed above (`extractRelations` -> `extractReplacementTypes` or something similar), then this test name can be changed too. Just a reminder. Just needs to be in sync.

File: ontology-common/src/test/java/uk/ac/ebi/quickgo/ontology/common/OntologyRepositoryIT.java [hunk 3]
```diff
@@ -105,9 +105,26 @@ public void retrieves2DocsCoreFieldsOnly() {
     }
```
Comment (eddturner) [inline ontology-common/src/test/java/uk/ac/ebi/quickgo/ontology/common/OntologyRepositoryIT.java]: probably originating from previous changes, but looks like this class could have its methods reordered according to visibility rules (intellij ctrl-shift-a, then type "rearrange", then <enter>). ... I noticed a private method higher than a public test method.

**Comment by eddturner** [comment]
Changes look good. Other than the comments I've made explicitly in code, I think the main thing is the name of `relations` throughout -- should this be `replacementTypes` or something similar? I'm not sure why it's `relations`.

**Comment by eddturner** [comment]
btw -- looks like master is broken for now, so this shouldn't be merged until that's fixed.

File: indexing/src/main/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverter.java (hunk 1 above)
Comment (rantunesboreas) [inline indexing/src/main/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverter.java]: Have created methods to make it easier to understand when to extract a replacement and when to extract a replaces.

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverterTest.java (hunk 2 above)
Comment (rantunesboreas) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverterTest.java]: extractRelation is used in extractReplacements and extractReplaces. Code is used in both case, hence the name being a bit more generic.

**Comment by rantunesboreas** [comment]
I have made your requested changes.

I have also decided to remove the replacedBy attribute from OBOTerm and OntologyDocument. seeing as it is already represented in the replacements section.

**Comment by eddturner** [comment]
cool. 

Just double checked tests before merging and i see there's a failure after a renamed variable.
E.g., see
`uk.ac.ebi.quickgo.index.ontology.converter.GenericTermToODocConverterTest#convertSimpleFields`

**Pull request merged**